client/components/OperationList.js

```diff
@@ -165,6 +165,142 @@ function ComputeAttachmentLink(op) {
+            );
+        }
+        return (
+            <input className="form-control"
+                type="text"
+                ref='customlabel'
+                id={this.props.operation.id}
+                defaultValue={this.props.defaultValue()}
+                onBlur={this.onBlur.bind(this)}
+                onKeyUp={this.onKeyUp.bind(this)}
+            />
+            );
+    }
+}
+
+class DetailedViewLabelComponent extends LabelComponent {
+    constructor(props) {
+        super(props);
+        has(props, 'operation');
+        this.defaultValue = this.defaultValue.bind(this);
+        this.buttonLabel = this.buttonLabel.bind(this);
+    }
+
+    buttonLabel() {
+        if (this.props.operation.customLabel === null || this.props.operation.customLabel.replace(/\ /g, '').length == 0){
+            return <em className="text-muted">{t('operations.add_custom_label') || "Add custom label"}</em>;
+        }
+        return <div className="label-button">{this.props.operation.customLabel}</div>;
+    }
+
+    defaultValue() {
+        if (this.props.operation.customLabel === null || this.props.operation.customLabel.replace(/\ /g, '').length == 0){
+            return this.props.label;
+        }
+        return this.props.operation.customLabel;
+    }
+
+    render() {
+        return <LabelComponent
+                    operation={this.props.operation}
+                    defaultValue={this.defaultValue}
+                    buttonLabel={this.buttonLabel} />;
+    }
+}
+
+class OperationListViewLabelComponent extends LabelComponent {
+    constructor(props) {
+        super(props);
+        has(props, 'operation');
+        has(props, 'label');
+        has(props, 'link');
+        this.defaultValue = this.defaultValue.bind(this);
+        this.buttonLabel = this.buttonLabel.bind(this);
+    }
+
+    buttonLabel() {
+        if (this.props.operation.customLabel === null || this.props.operation.customLabel.replace(/\ /g, '').length == 0){
+            return <div className="label-button text-uppercase">{this.props.label}</div>;
+        }
+        return <div className="label-button text-uppercase">{this.props.operation.customLabel}</div>;
+    }
+
+    defaultValue() {
+        if (this.props.operation.customLabel === null || this.props.operation.customLabel.replace(/\ /g, '').length == 0){
+            return this.props.label;
+        }
+        return this.props.operation.customLabel;
+    }
+
+    render() {
+        if (typeof this.props.link !== 'undefined') {
+            return <div className="input-group">{this.props.link}<LabelComponent
+                    operation={this.props.operation}
+                    defaultValue={this.defaultValue}
+                    buttonLabel={this.buttonLabel} /></div>;
+        }
+        return <LabelComponent
+                    operation={this.props.operation}
+                    defaultValue={this.defaultValue}
+                    buttonLabel={this.buttonLabel} />;
```

Comment:
With inheritance, can you instead use `let comp = super.render();` or `let comp = super();`? This should prevent the copying/pasting.